### PR TITLE
Devel

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
This is initial release 0.1.dev0.

Major Additions include:

* Integration of Message Infrasture
* Addition of persistent state tracking using local database
* Maturation of CI infrastructure and docker compose files
* Separation of Executor and message handler and agent classes to define a clear communication strategy
* Switch to RabbitMQ from NATS

### CODE REVIEW

**Comment by renan-souza** [comment]
@par-hermes format

**Comment by renan-souza** [comment]
@par-hermes format

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,13 +1,96 @@
-# Pull Request Guidelines
+# Project Guidelines
```
Comment (ketancmaheshwari) [inline CONTRIBUTING.md:1]: Some of these are not practical for small / medium sized teams. A few others are not practical for a scientific software.

**Review by ketancmaheshwari — DISMISSED** [review]
Too much extraneous code making the codebase bulky. Very unpythonic.

File: examples/imagemagick_files.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2022 Oak Ridge National Laboratory.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the MIT License.
+
```
[the remark this answers is not in the document]
Comment (JoshuaSBrown) [reply examples/imagemagick_files.py:8]: Do you have a better suggestion? imagemagick_shell_workflow.py ?

File: CONTRIBUTING.md (hunk 1 above)
Comment (renan-souza) [reply CONTRIBUTING.md:1]: @ketancmaheshwari Can you please be more specific? Which recommendation you think is not practical and why?

File: tests/test_file_uri_separator.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
+# Local imports
+from zambeze.orchestration.plugin_modules.file_uri_separator import FileURISeparator
+
+# Standard imports
+import pytest
+import os
+
```
[the remark this answers is not in the document]
Comment (JoshuaSBrown) [reply tests/test_file_uri_separator.py:7]: That is a good suggestion, a decorator sets up the defaults, I have seen this before.

File: tests/test_plugin_rsync_message_validator.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# Local imports
+# fmt: off
```
[the remark this answers is not in the document]
Comment (JoshuaSBrown) [reply tests/test_plugin_rsync_message_validator.py:2]: Can you be a little more specific, I'm not sure what you are referring too?

File: zambeze/orchestration/plugin_modules/common_dataclasses.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
+from dataclasses import dataclass, field
+
+
```
[the remark this answers is not in the document]
Comment (renan-souza) [reply zambeze/orchestration/plugin_modules/common_dataclasses.py:3]: @ketancmaheshwari This is the official documentation: https://docs.python.org/3/library/dataclasses.html

Please let us know if you need more clarification other than the official documentation.

File: zambeze/orchestration/message/general_message_components.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
+"""These items are required as part of any message
+
```
[the remark this answers is not in the document]
Comment (JoshuaSBrown) [reply zambeze/orchestration/message/general_message_components.py:2]: I generally find it helpful to organize objects into a folder hierarchy where the items higher in the hierarchy are useful to code in child folders. This is such as case. I will try to add some comments to make this clearer.

File: examples/imagemagick_files.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (JoshuaSBrown) [reply examples/imagemagick_files.py:8]: Are you talking about each of the example files or are you talking about every single source file?

File: zambeze/orchestration/message/activity_message/message_activity.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import logging
+from ..abstract_message import AbstractMessage
+from zambeze.orchestration.zambeze_types import MessageType
+from typing import Optional
+from dataclasses import dataclass
+
+
```
[the remark this answers is not in the document]
Comment (JoshuaSBrown) [reply zambeze/orchestration/message/activity_message/message_activity.py:7]: These names are non ideal. I don't really have good alternatives though.

**Review by rafaelfsilva — APPROVED** [review]
I have checked most of the files during the review process these past 2 weeks, and this current version satisfies the need for Zambeze's v0.1.

**Pull request merged**